app/models/concerns/race_and_ethnicity_calculations.rb

```diff
@@ -0,0 +1,95 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module RaceAndEthnicityCalculations
+  extend ActiveSupport::Concern
+
+  def race_ethnicities_breakdowns(scope)
+    race_ethnicities_breakdown = race_ethnicity_combinations.values.map { |m| [m, Set[]] }.to_h
+
+    race_data = pluck_to_hash((['id'] + race_col_lookup.keys.excluding('race_none')).map { |k, v| [k, v] }.to_h, scope)
+    race_ethnicity_combinations.each do |key, value|
+      race_ethnicities_breakdown[value] = lookups[key]&.call(race_data)
+    end
+    race_ethnicities_breakdown
+  end
+
+  def race_col_lookup
+    HudUtility2024.races.map { |k, _| [k.underscore, k] }.to_h
+  end
+
+  private def pluck_to_hash(columns, scope)
+    scope.pluck(*columns.keys).map do |row|
+      Hash[columns.keys.zip(row)]
+    end
+  end
+
+  private def single_race(race_data, race)
+    [].tap do |ids|
+      race_data.select { |enrollment| enrollment[race] && !enrollment['hispanic_latinaeo'] }.
+        each { |e| ids << e['id'] if e.excluding('id', 'hispanic_latinaeo').count { |m| m.count(true) == 1 } == 1 }
+    end
+  end
+
+  private def single_race_latinaeo(race_data, race)
+    [].tap do |ids|
+      race_data.select { |enrollment| enrollment[race] && enrollment['hispanic_latinaeo'] }.
+        each { |e| ids << e['id'] if e.excluding('id', 'hispanic_latinaeo').count { |m| m.count(true) == 1 } == 1 }
+    end
+  end
+
+  private def only_hispanic_latinaeo(race_data)
+    [].tap do |ids|
+      race_data.select { |enrollment| enrollment['hispanic_latinaeo'] }.
+        each { |e| ids << e['id'] if e.excluding('id').count { |m| m.count(true) == 1 } == 1 }
+    end
+  end
+
+  private def multi_racial(race_data)
+    [].tap do |ids|
+      race_data.reject { |enrollment| enrollment['hispanic_latinaeo'] }.
+        each { |e| ids << e['id'] if e.excluding('id', 'hispanic_latinaeo').count { |m| m.count(true) == 1 } > 1 }
+    end
+  end
+
+  private def multi_racial_with_hispanic_latinaeo(race_data)
+    [].tap do |ids|
+      race_data.select { |enrollment| enrollment['hispanic_latinaeo'] }.
+        each { |e| ids << e['id'] if e.excluding('id', 'hispanic_latinaeo').count { |m| m.count(true) == 1 } > 1 }
+    end
+  end
+
+  private def race_none(race_data)
+    [].tap do |ids|
+      race_data.each { |e| ids << e['id'] if e.except('id').values.all?(false) }
+    end
+  end
+
+  private def lookups
```

Comment:
Given that this is in a concern now, I think we should rename `lookups` to something like `race_and_ethnicity_lookups` to decrease collisions with existing methods.